test/engines/misc/sends/sends.t.cpp

```diff
@@ -0,0 +1,242 @@
+#include <algorithm>
+#include <ostream>
+
+#include "../test/graphics.t.hpp"
+#include "core/ui/nvg/Canvas.hpp"
+#include "core/ui/nvg/Color.hpp"
+#include "core/ui/vector_graphics.hpp"
+#include "itc/action_sender.hpp"
+#include "itc/prop.hpp"
+#include "testing.t.hpp"
+#include "util/algorithm.hpp"
+
+using namespace otto::test;
+using namespace otto::core::props;
+
+using Sender = otto::itc::DirectActionSender<>;
+
+struct Props {
+  Sender sender;
+
+  Sender::Prop<struct dry_tag, float> dry = {sender, 1, limits(0, 1), step_size(0.01)};
+  Sender::Prop<struct fx1_tag, float> fx1 = {sender, 0, limits(0, 1), step_size(0.01)};
+  Sender::Prop<struct fx2_tag, float> fx2 = {sender, 0, limits(0, 1), step_size(0.01)};
+
+  struct StoredLevels {
+    float fx1 = 0;
+    float fx2 = 0;
+    float dry = 1;
+    float s = 1;
+  } stored_levels;
+};
+
+bool operator==(const Props& lhs, const std::tuple<float, float, float>& rhs)
+{
+  return lhs.fx1 == std::get<0>(rhs) && lhs.fx2 == std::get<1>(rhs) && lhs.dry == std::get<2>(rhs);
+}
+std::ostream& operator<<(std::ostream& s, const Props& v)
+{
+  return s << "[" << v.fx1 << ", " << v.fx2 << ", " << v.dry << "]";
+}
+
+struct Sendstest {
+  Sendstest()
```

Comment:
maybe take a look at using the actual Sends engine with this. If thats not possible, never mind, and you should probably remove the tests.